src/main/scala/is/hail/expr/ir/PruneDeadFields.scala

```diff
@@ -329,6 +329,12 @@ object PruneDeadFields {
               memo)
           case None => memoizeTableIR(child, requestedType, memo)
         }
+      case LocalizeEntries(child, fieldName) =>
+        val minChild = minimal(child.typ)
+        val childDep = minChild.copy(
+          globalType = requestedType.globalType,
+          rvRowType = unify(child.typ.rvRowType, minChild.rvRowType, requestedType.rowType))
```

Comment:
needs to pull out / rename the field as we discussed. Also add a rebuild rule that looks at the requested type again to see if the localized array field is requested, and rebuilds as `MatrixRowsTable` if it's not.